net/ampr-ripd/files/ampr-ripd-init

```diff
@@ -0,0 +1,87 @@
+#!/bin/sh /etc/rc.common
+
+START=95
+STOP=10
+
+EXTRA_COMMANDS="configure"
+EXTRA_HELP="        configure       Configure service parameters"
+
+start() {
+	default_addr="44.127.254.254"
+	if [ "$(uci get network.amprlan.ipaddr)" = "$default_addr" ] || \
+		[ "$(uci get network.amprwan.ipaddr)" = "$default_addr" ]; then
+
+		cat <<-EOF
+		ampr-ripd is not fully configured.
+		You must run /etc/init.d/ampr-ripd configure.
+		EOF
+
+		exit 1
+	fi
+	if [ ! -d /var/lib/ampr-ripd ]; then
+		mkdir -p /var/lib/ampr-ripd
+	fi
+	ip tunnel change ttl 64 mode ipip tunl0
+	ip link set dev tunl0 up
+	ifconfig tunl0 mtu 1480
+	tunnet=$(uci -q get ampr-ripd.network.tunnet)
+	/usr/sbin/ampr-ripd -s -r -t 44 -i tunl0 -a "$tunnet"
+}
+
+stop() {
+	ifconfig tunl0 down
+	killall ampr-ripd
+}
+
+configure() {
+	if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
+		cat <<-EOF
+
+		Usage: /etc/init.d/ampr-ripd configure [amprhost] [amprmask] [amprnet]
+
+		       amprhost is the 44 net address assigned to your OpenWrt host
+		       amprmask is the full netmask of your AMPR assigned network
+		       amprnet is the network number of your AMPR assigned network 
+
+		   Eg. /etc/init.d/ampr-ripd configure 44.127.254.1 255.255.255.0 44.127.254.0
```

Comment:
Thinking about this. Shouldnt the binary help you with that? I think you are trying reinvent a wheel here, which leads to be things really confusing and complicated. They don't need to be like that.

Reply:
I'm not sure I follow: what should the binary help with in this case?

Reply:
I mean the resulting binary on the router itself. It should not start if it is not configured or also it should provide ways how to provide it, if you want to have it more user-friendly.

E.g. /usr/sbin/ampr-ripd configure or something like it. 

But still, in my point of view, this needs to be done differently, because it does so *magical* things to do network, which is not so good. I can not imagine situation that I will install this package and it tweaks my network so much/so badly (adding default rules, network interfaces, firewall zones, etc) and then removing it. What if something goes bad and that it could not be removed, added?

It is done so complicated. 🤷

Reply:
I see where you're coming from. This seems like quite a specialized package, so perhaps potential users won't mind.

As an aside, have you seen a similar approach to changing multiple system-wide settings in the open Tailscale PR? I don't want to link it here to create noise.

Reply:
Unfortunately, not yet.